File: server/games/love-letter/agent.js
import IAgent from "../i-agent.js";
import Action from "./action.js";

class Agent extends IAgent {

    index;
    state;

    constructor(token) {
        super(token);
    }

    newRound(state) {
        this.state = state;
    }

    see(action, state) {
        this.state = state;
    }

    async playCard(card) {
        const action = await this.awaitEvent()
        if (action === null) return null;
        return Action[action.action](...action.params);
    }

    /**
     * @returns The state corresponding to this agent.
     */
    getState() {
        const agentState = {...this.state};
        delete agentState.agents;
        return agentState;
    }
    
}

export default Agent;

File: client/agents/python/love_letter/agent.py
from love_letter.agent_io import AgentIO

class Agent:
    def __init__(self, token, server):
        self.agentIO = AgentIO(token, server)
        self.token = token
        self.state = None
        
    def updateState(self):
        self.state = self.agentIO.getState()
    
    def move(self):
        move = None

        return move